core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -601,6 +620,10 @@ public AccessToken getAccessToken(String code, URI callbackUri)
             assertAuthenticateOAuth2Bearer(client, getManagementLocation(baseUri), "http://example.com/authorize");
             assertResponseCode(client, getInternalLocation(baseUri), SC_FORBIDDEN);
 
+            // We must add the nonce to the ID token we will soon generate.
+            idTokenBuilder.claim(NONCE, hashNonce(bearer.getNonceCookie().getValue()));
+            // The second call to `assertAuthenticateOAuth2Bearer` above overwrites the nonce cookie we need.
```

Comment:
In this case this test should be split into two separate tests, as the setup changes during the test.

Reply:
I don't understand how this test would be split in two and what that would accomplish. The reason for this change is that ID tokens are now validated against the nonce cookie which never happened before. This changes nothing about the actual test, it's just additional setup necessary to execute the existing test.